src/compiler/sourcemap.ts

```diff
@@ -573,7 +583,8 @@ namespace ts {
     }
 
     function compareSourcePositions(left: SourceMappedPosition, right: SourceMappedPosition) {
-        return compareValues(left.sourceIndex, right.sourceIndex);
+        Debug.assert(left.sourceIndex === right.sourceIndex);

```

Comment:
I'd comment that this is only true so long as we continue to pre-group all mappings by source index - otherwise we'd need to sort by both.